elements/pf-popover/test/pf-popover.spec.ts

```diff
@@ -126,38 +158,71 @@ describe('<pf-popover>', function() {
           role: 'button',
         },
       ]);
-    fixtureCleanup();
-    const container = await fixture(html`
-      <div>
-        <pf-popover
-          id="popover"
-          trigger="btn-1"
-          heading="Popover heading"
-          body="Popovers are triggered by click rather than hover."
-          footer="Popover footer"
-        >
-        </pf-popover>
-        <button id="btn-1">Toggle popover 1</button>
-        <button id="btn-2">Toggle popover 2</button>
-      </div>
-    `);
-    snapshot = await a11ySnapshot();
-    const popover: PfPopover | null = container.querySelector('pf-popover');
-    const btn1: HTMLButtonElement | null = container.querySelector('#btn-1');
-    const btn2: HTMLButtonElement | null = container.querySelector('#btn-2');
-    expectClose();
-    btn1?.click();
-    snapshot = await a11ySnapshot();
-    expectOpen();
-    // Close the popover
-    await sendKeys({ press: 'Enter' });
-    // Update trigger element
-    popover?.setAttribute('trigger', 'btn-2');
-    btn1?.click();
-    snapshot = await a11ySnapshot();
-    expectClose();
-    btn2?.click();
-    snapshot = await a11ySnapshot();
-    expectOpen();
+    }
+
+    beforeEach(async function() {
+      resetMouse();
+      fixtureCleanup();
+      const container = await fixture(html`
+        <div>
+          <pf-popover
+            id="popover"
+            trigger="btn-1"
+            heading="Popover heading"
+            body="Popovers are triggered by click rather than hover."
+            footer="Popover footer"
+          >
+          </pf-popover>
+          <button id="btn-1">Toggle popover 1</button>
+          <button id="btn-2">Toggle popover 2</button>
+        </div>
+      `);
+      element = container.querySelector('pf-popover')!;
+      btn1 = container.querySelector('#btn-1')!;
+      btn2 = container.querySelector('#btn-2')!;
+    });
+
+    it('starts closed', expectClose);
+    describe('clicking the trigger', function() {
+      beforeEach(async function() {
+        await element.updateComplete;
+        await clickElementCenter(btn1);
+        await element.updateComplete;
+      });
+      it('shows the popover', expectOpen);
+      describe('then setting the trigger to the sibling button', function() {
+        beforeEach(async function() {
+          await element.updateComplete;
+          // Update trigger element
+          element.setAttribute('trigger', 'btn-2');
+          await element.updateComplete;
+        });
+        describe('clicking the first button', function() {
+          beforeEach(async function() {
+            await element.updateComplete;
+            await clickElementCenter(btn1);
+            await element.updateComplete;
+          });
+          it('remains closed', expectClose);
+        });
+        describe('clicking the sibling button', function() {
+          beforeEach(async function() {
```

Comment:
This test is failing after I made the switch from `show` to `toggle` on the event handlers we attached to the trigger. The sequence of events looks correct to me (btw, very cool how you wrote these tests!). If I change this hook to `before` instead of `beforeEach`, all tests pass 🤔 but I'm not sure why!?